packet/models.go

```diff
@@ -106,18 +106,61 @@ func NewDiscovery(b []byte) (Discovery, error) {
 }
 
 // Instance models the instance data as returned by the API
-type Instance struct {
+type Instance interface {
```

Comment:
I don't think this is a good route to take since this is exporting implementation details to callers and making all this more complicated. I think the easiest option is to have an unexported `instance` struct that has both `OperatingSystem` and `OperatingSystemVersion` and then we grab the data from the correct field depending which mode we're running in and then convert to `Instance`.

Reply:
That makes sense, but how do I grab the data and convert to Instance? Is `Instance` an empty interface?

Reply:
Copying fields one-by-one. That sucks but the suck is contained into just cacher code that will go away soon anyway.